src/lib/services/state-service.ts

```diff
@@ -697,7 +704,50 @@ export default class StateService {
         );
     }
 
-    async export({
+    async export(opts: IExportIncludeOptions): Promise<{
+        features: FeatureToggle[];
+        strategies: IStrategy[];
+        version: number;
+        projects: IProject[];
+        tagTypes: ITagType[];
+        tags: ITag[];
+        featureTags: IFeatureTag[];
+        featureStrategies: IFeatureStrategy[];
+        environments: IEnvironment[];
+        featureEnvironments: IFeatureEnvironment[];
+    }> {
+        if (this.flagResolver.isEnabled('variantsPerEnvironment')) {
+            return this.exportV4(opts);
+        }
+        // adapt v4 to v3
+        const v4 = await this.exportV4({ ...opts, includeEnvironments: true });
+        if (opts.includeFeatureToggles || true) {
+            const keepEnv = v4.environments
+                .filter((env) => env.type === 'production' && env.enabled)
+                .sort((e1, e2) => e2.sortOrder - e1.sortOrder)[0]; // TODO test order
```

Comment:
We're assuming we have environments, should we be more defensive here. I realise our discussion before lunch only covered the happy path.

Reply:
That's a good point and also reminds me that I should validate the order is correct

Reply:
shoot, my answer was going to be: no, this statement is only executed when includeFeatureToggles is `undefined` or true, but after looking at the code I see an error in my logic :see_no_evil: I'll fix the condition